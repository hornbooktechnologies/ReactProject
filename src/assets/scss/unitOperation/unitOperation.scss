@import "../_color";
@import "../_variables";

.bg-grey-blue-40-a10 {
  background-color: rgba($grey_blue_40, 0.1);
}
.color-grey-blue-60 {
  color: $grey_blue_60;
}
.min-w-1190 {
  min-width: 1190px;
}
.m-w-110 {
  min-width: 110px;
}
.max-height-500 {
  max-height: $val10 * 50;
}
.font-weight-500 {
  font-weight: 500;
}
// uk Table start
.uk-table {
  &.uk-table-divider {
    th,
    td {
      border: 1px $solid $grey_blue_40;
      text-align: left;
      padding: $val10 $val10 - 2;
      color: $white;
      font-size: $val15 - 1;
      .icn {
        position: $relative;
        top: 2px;
        &.icn-right {
          float: $right;
        }
      }
    }
    th {
      background-color: rgba($white, 0.1);
      font-size: $val15 + 1;
      vertical-align: $middle;
      text-transform: $inherit;
      font-weight: 500;
      text-align: $center;
    }
  }
  &.uk-text-center {
    th,
    td {
      text-align: $center;
    }
  }
  &.uk-text-right {
    th,
    td {
      text-align: $right;
    }
  }
  &.uk-table-sm {
    th {
      font-size: $val15 - 2 !important;
    }
    td {
      font-size: $val15 - 3 !important;
    }
  }
  &.uk-table-middle {
    th,
    td {
      vertical-align: $middle !important;
    }
  }
}
.min-w-86 {
  min-width: 86px;
  width: fit-content;
  margin-top: $val0;
}

@media (max-width: 767px) {
  .media-mt-30 {
    margin-top: 3 * $val10;
  }
}

