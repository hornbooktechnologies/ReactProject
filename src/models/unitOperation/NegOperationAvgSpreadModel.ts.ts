export type NegOperationAvgSpreadModel = {negOperationAvrgSpread:AverageSpreadCardModel[]}
export type NegativeOperationAvgSpreadModel = AverageSpreadCardModel[]

export type AverageSpreadCardModel = {
  title: string,
  Suffix: string,
  plan: AverageSpreadRecordModel
  actual_forcast?: AverageSpreadRecordModel
  yearStartToPresent?: AverageSpreadRecordModel
}

export type AverageSpreadRecordModel = {
  header?:string | undefined
  previousValue?: number | undefined | string
  suffix?: string 
  value?: number | undefined | string
}
