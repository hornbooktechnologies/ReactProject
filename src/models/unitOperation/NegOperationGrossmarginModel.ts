  export type NegGrossmarginTableModel = negativeGrossMargin
  export type NegativeGrossMarginTableModel = negativeGrossMargin
  export type NegativeGrossMarginModel = {negativeGrossMargin:negativeGrossMargin}
  export type negativeGrossMargin = {
    actualOrForcastAnnual:number|null|undefined|string
    actualOrForcastEnd:number|null|undefined|string
    actualOrForcastPresent:number|null|undefined|string
    impactOnEBITDAAnnual:number|null|undefined|string
    impactOnEBITDAEnd:number|null|undefined|string
    impactOnEBITDAPresent:number|null|undefined|string
    planAnnual:number|null|undefined|string
    planEnd:number|null|undefined|string
    planPresent:number|null|undefined|string
    previousActualOrForcastAnnual:number|null|undefined|string
    previousActualOrForcastEnd:number|null|undefined|string
    previousActualOrForcastPresent:number|null|undefined|string
    previousImpactOnEBITDAAnnual:number|null|undefined|string
    previousImpactOnEBITDAEnd:number|null|undefined|string
    previousImpactOnEBITDAPresent:number|null|undefined|string
    previousPlanAnnual:number|null|undefined|string
    previousPlanEnd:number|null|undefined|string
    previousPlanPresent:number|null|undefined|string
    customClassTop: string|undefined
    customClassEnd: string|undefined
    prefix:string|undefined
    suffix:string|undefined
}
