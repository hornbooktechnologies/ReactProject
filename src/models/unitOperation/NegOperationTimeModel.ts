export type NegOperationTimeModel =  {
  header: string ,
  plan:number | string,
  actualOrForecast:number | string,
  previousPlan?:number | undefined | null | string,
  previousActualOrForecast?:number | undefined | null | string,
  suffix?:string | undefined | null,
  titleX?:string | undefined | null,
  titleY?:string | undefined | null,
}
export type NegativeOperationTimeModel =  NegOperationTimeModel[]
export type NegOperationTimeDonutChartModel = {negOperationTime :NegOperationTimeModel[] }

