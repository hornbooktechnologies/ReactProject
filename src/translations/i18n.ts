import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
