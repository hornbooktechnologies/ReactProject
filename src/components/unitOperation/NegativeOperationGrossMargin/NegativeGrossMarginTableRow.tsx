import React from 'react'
import { NegativeGrossMarginModel } from '../../../models/unitOperation/NegOperationGrossmarginModel'
import i18n from '../../../translations/i18n'
import { cssClassName } from '../../../utils/CssConstants'
import { addCustomClass, getDiffOfValues, IstextHighlight, setTooltipPosition } from '../../../utils/utils'
import Tooltip from '../../common/Tooltip'

const NegativeGrossMarginTableRow = ({negativeGrossMargin}:NegativeGrossMarginModel) => {
  let diffImpactAnnualVal = getDiffOfValues(negativeGrossMargin?.impactOnEBITDAAnnual,negativeGrossMargin?.previousImpactOnEBITDAAnnual)
  let diffImpactPresentVal = getDiffOfValues(negativeGrossMargin?.impactOnEBITDAPresent,negativeGrossMargin?.previousImpactOnEBITDAPresent)
  let diffImpactEndVal = getDiffOfValues(negativeGrossMargin?.impactOnEBITDAEnd,negativeGrossMargin?.previousImpactOnEBITDAEnd)
  let diffPlanAnnualVal = getDiffOfValues(negativeGrossMargin?.planAnnual,negativeGrossMargin?.previousPlanAnnual)
  let diffPlanPresentVal = getDiffOfValues(negativeGrossMargin?.planPresent,negativeGrossMargin?.previousPlanPresent)
  let diffPlanEndVal = getDiffOfValues(negativeGrossMargin?.planEnd,negativeGrossMargin?.previousPlanEnd)
  let diffActualOrForcastAnnualVal = getDiffOfValues(negativeGrossMargin?.actualOrForcastAnnual,negativeGrossMargin?.previousActualOrForcastAnnual)
  let diffActualOrForcastPresentVal = getDiffOfValues(negativeGrossMargin?.actualOrForcastPresent,negativeGrossMargin?.previousActualOrForcastPresent)
  let diffActualOrForcastEndVal =  getDiffOfValues(negativeGrossMargin?.actualOrForcastEnd,negativeGrossMargin?.previousActualOrForcastEnd)

  return (
    <>
      <tr>
        <th className={`${cssClassName.BG_GREY_BLUE_40_A10}`} colSpan={2}>
         {i18n.t("UNIT_OPERATION.IMPACT_SPREAD_EBITDA")} 
        </th>
        <td className={` ${addCustomClass(diffImpactAnnualVal,negativeGrossMargin?.customClassEnd) } uk-text-middle`} >
          <span className={`${cssClassName.COLOR_GREY_BLUE_60} font-12`}>
          {negativeGrossMargin?.prefix}({negativeGrossMargin?.planAnnual} - {negativeGrossMargin?.actualOrForcastAnnual}){negativeGrossMargin?.suffix}
          </span>
          <span  className={` ${IstextHighlight(diffImpactAnnualVal)} font-14 uk-display-block`}>{negativeGrossMargin?.prefix}{negativeGrossMargin?.impactOnEBITDAAnnual}&nbsp;{negativeGrossMargin?.suffix}
        </span>
         {diffImpactAnnualVal === true ? (
           <Tooltip
           prefix={negativeGrossMargin?.prefix}
           previousDayValue={negativeGrossMargin?.previousImpactOnEBITDAAnnual}
           unit={negativeGrossMargin?.suffix}
           customClass={setTooltipPosition(negativeGrossMargin?.customClassEnd)}
         />
       ) : null}
        </td>
        <td className={` ${addCustomClass(diffImpactPresentVal,negativeGrossMargin?.customClassEnd) } uk-text-middle`} >
          <span className={`${cssClassName.COLOR_GREY_BLUE_60} font-12`}>
          {negativeGrossMargin?.prefix}({negativeGrossMargin?.planPresent} - {negativeGrossMargin?.actualOrForcastPresent}){negativeGrossMargin?.suffix}
          </span>
           <span  className={` ${IstextHighlight(diffImpactPresentVal)} font-14 uk-display-block`}>
          {negativeGrossMargin?.prefix}{negativeGrossMargin?.impactOnEBITDAPresent}&nbsp;{negativeGrossMargin?.suffix}
          </span>
          {diffImpactPresentVal === true ? (
           <Tooltip
           prefix={negativeGrossMargin.prefix}
           previousDayValue={negativeGrossMargin.previousImpactOnEBITDAPresent}
           unit={negativeGrossMargin.suffix}
           customClass={setTooltipPosition(negativeGrossMargin.customClassEnd)}
         />
       ) : null}
        </td>
        <td className={` ${addCustomClass(diffImpactEndVal,negativeGrossMargin?.customClassEnd) } uk-text-middle`} >
          <span className={`${cssClassName.COLOR_GREY_BLUE_60} font-12`}>
          {negativeGrossMargin?.prefix}({negativeGrossMargin?.planEnd} - {negativeGrossMargin?.actualOrForcastEnd}){negativeGrossMargin?.suffix}
          </span>
           <span  className={` ${IstextHighlight(diffImpactEndVal)} font-14 uk-display-block`}>
          {negativeGrossMargin?.prefix}{negativeGrossMargin?.impactOnEBITDAEnd}&nbsp;{negativeGrossMargin?.suffix}
          </span>
          {diffImpactEndVal === true ? (
           <Tooltip
           prefix={negativeGrossMargin.prefix}
           previousDayValue={negativeGrossMargin.previousImpactOnEBITDAEnd}
           unit={negativeGrossMargin.suffix}
           customClass={setTooltipPosition(negativeGrossMargin.customClassEnd)}
         />
       ) : null}
        </td>
      </tr>
      <tr>
        <th
          className={`${cssClassName.BG_GREY_BLUE_40_A10} w-220`}
          rowSpan={2}
        >
         {i18n.t("UNIT_OPERATION.GROSSMARGIN_SPREAD")} 
        </th>
        <th className={`${cssClassName.BG_GREY_BLUE_40_A10} w-220`}>
        {i18n.t("LABELS.PLAN")} 
        </th>
        <td className={`${addCustomClass(diffPlanAnnualVal,negativeGrossMargin?.customClassTop) } uk-text-middle`}>
        <span className={`${IstextHighlight(diffPlanAnnualVal)} uk-display-block`}>
          {negativeGrossMargin?.prefix}{negativeGrossMargin?.planAnnual}&nbsp;{negativeGrossMargin?.suffix}
        </span>
        {diffPlanAnnualVal === true ? (
           <Tooltip
           prefix={negativeGrossMargin.prefix}
           previousDayValue={negativeGrossMargin.previousPlanAnnual}
           unit={negativeGrossMargin.suffix}
           customClass={setTooltipPosition(negativeGrossMargin.customClassTop)}
         />
       ) : null}
        </td>
        <td className={`${addCustomClass(diffPlanPresentVal,negativeGrossMargin?.customClassTop) } uk-text-middle`}>
        <span className={`${IstextHighlight(diffPlanPresentVal)} uk-display-block`}>
          {negativeGrossMargin?.prefix}{negativeGrossMargin?.planPresent}&nbsp;{negativeGrossMargin?.suffix}
        </span>
        {diffPlanPresentVal === true ? (
           <Tooltip
           prefix={negativeGrossMargin.prefix}
           previousDayValue={negativeGrossMargin.previousPlanPresent}
           unit={negativeGrossMargin.suffix}
           customClass={setTooltipPosition(negativeGrossMargin.customClassTop)}
         />
       ) : null}
        </td>
        <td className={`${addCustomClass(diffPlanEndVal,negativeGrossMargin?.customClassTop) } uk-text-middle`}>
        <span className={`${IstextHighlight(diffPlanEndVal)} uk-display-block`}>
          {negativeGrossMargin?.prefix}{negativeGrossMargin?.planEnd}&nbsp;{negativeGrossMargin?.suffix}
        </span>
        {diffPlanEndVal === true ? (
           <Tooltip
           prefix={negativeGrossMargin.prefix}
           previousDayValue={negativeGrossMargin.previousPlanEnd}
           unit={negativeGrossMargin.suffix}
           customClass={setTooltipPosition(negativeGrossMargin.customClassTop)}
         />
       ) : null}
        </td>
      </tr>
      <tr>
        <th className={`${cssClassName.BG_GREY_BLUE_40_A10}`}>
          {i18n.t("LABELS.ACTUAL_FORECAST")} 
        </th>
        <td className={`${addCustomClass(diffActualOrForcastAnnualVal,negativeGrossMargin?.customClassTop) } uk-text-middle`}>
        <span className={`${IstextHighlight(diffActualOrForcastAnnualVal)} uk-display-block`}>
          {negativeGrossMargin?.prefix}{negativeGrossMargin?.actualOrForcastAnnual}&nbsp;{negativeGrossMargin?.suffix}
          </span>
          {diffActualOrForcastAnnualVal === true ? (
           <Tooltip
           prefix={negativeGrossMargin.prefix}
           previousDayValue={negativeGrossMargin.previousActualOrForcastAnnual}
           unit={negativeGrossMargin.suffix}
           customClass={setTooltipPosition(negativeGrossMargin.customClassTop)}
         />
       ) : null}
        </td>
        <td className={`${addCustomClass(diffActualOrForcastPresentVal,negativeGrossMargin?.customClassTop) } uk-text-middle`}>
        <span className={`${IstextHighlight(diffActualOrForcastPresentVal)} uk-display-block`}>{negativeGrossMargin?.prefix}{negativeGrossMargin?.actualOrForcastPresent}&nbsp;{negativeGrossMargin?.suffix}</span>
          {diffActualOrForcastPresentVal === true ? (
           <Tooltip
           prefix={negativeGrossMargin.prefix}
           previousDayValue={negativeGrossMargin.previousActualOrForcastPresent}
           unit={negativeGrossMargin.suffix}
           customClass={setTooltipPosition(negativeGrossMargin.customClassTop)}
         />
       ) : null}
        </td>
        <td className={`${addCustomClass(diffActualOrForcastEndVal,negativeGrossMargin?.customClassTop) } uk-text-middle`}>
        <span className={`${IstextHighlight(diffActualOrForcastEndVal)} uk-display-block`}>
          {negativeGrossMargin?.prefix}{negativeGrossMargin?.actualOrForcastEnd}&nbsp;{negativeGrossMargin?.suffix}
          </span>
          {diffActualOrForcastEndVal === true ? (
           <Tooltip
           prefix={negativeGrossMargin.prefix}
           previousDayValue={negativeGrossMargin.previousActualOrForcastEnd}
           unit={negativeGrossMargin.suffix}
           customClass={setTooltipPosition(negativeGrossMargin.customClassTop)}
         />
       ) : null}
          </td>
      </tr>
    </>
  )
}

export default NegativeGrossMarginTableRow
