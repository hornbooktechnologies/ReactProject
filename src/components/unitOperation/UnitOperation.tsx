import React, { useEffect, useState } from "react";
import { useTranslation } from 'react-i18next';
import { Kpi_005ApiRes, UnitOperationBaseModel } from "../../models/unitOperation/UnitOperationModel";
import { unitOperationApiMockData, unitOperationMockData } from "../../models/unitOperation/_mockData";
import LoadingError from "../common/LoadingError";
import PageHeader from "../common/PageHeader";
import Spinner from "../common/Spinner";
import AmountOfOperatingTime from "./amountOfOperatingTime/AmountOfOperatingTime";
import AverageSpread from "./averageSpread/AverageSpread";
import SpreadNumberOfCases from "./spreadNumberOfCases/SpreadNumberOfCases";
import { spreadOperationMockData } from "../../models/unitOperation/_mockData";
import NegativeOperationGrossMargin from "./NegativeOperationGrossMargin/NegativeGrossMarginTable";
import ReferenceSpecificationOfSpreadOperation from "./specificationSpread/ReferenceSpecificationOfSpreadOperation";
import SpecificationSpread from "./specificationSpread/SpecificationSpread";
import Top20DurationOfSpread from "./top20DurationOfSpread/Top20DurationOfSpread";
import { getCutoffFractionTime } from "../../utils/utils";
import { negOperationAvrgSpreadApiResponse, negOperationGrossmarginApiResponse, negOperationTimeApiResponse } from "../../utils/unitOperation/UnitOperationUtils";
import { NegativeOperationTimeModel} from "../../models/unitOperation/NegOperationTimeModel";
import { NegativeOperationAvgSpreadModel } from "../../models/unitOperation/NegOperationAvgSpreadModel.ts";
import { NegativeGrossMarginTableModel } from "../../models/unitOperation/NegOperationGrossmarginModel";
import "../../assets/scss/annualTarget/_annualstatus.scss";
import '../../assets/scss/unitOperation/unitOperation.scss';
import { DpmService } from "../../services/DpmServices";
import { ApiParamUtils } from "../../utils/apiUtils/ApiParamUtils";

const UnitOperation = () => {
  const { i18n } = useTranslation();
  const [unitOperationData, setUnitOperationData] = useState({} as UnitOperationBaseModel);
  const [negOperationGrossMarginData, setNegOperationGrossMargin] = useState({} as NegativeGrossMarginTableModel)
  const [negOperationTimeData, setNegOperationTime] = useState([] as NegativeOperationTimeModel)
  const [negOperationAvrgSpreadData, setNegOperationAvrgSpread] = useState([] as NegativeOperationAvgSpreadModel)
  const [unitOperationApiData, setUnitOperationApiData] = useState({});
  const [errorThrown, seterrorThrown] = useState(null)
  const [kpiResponse, setKpiResponse] = useState(false)
  const [time, setNoDataDate] = useState(
    getCutoffFractionTime()
  )

  const noDataHandler = (val: number, selectValue: string) => {
    setNoDataDate(val)
  }
    
  useEffect(() => {
    DpmService.fetchKPI_005Data(ApiParamUtils.plantId(),ApiParamUtils.unitCode()) 
    .then((res)=>{
      res=res.data
      if (res.Success) {
        setKpiResponse(res.Success)
        let kpiResponse = res.Data as Kpi_005ApiRes;
        let mappedNegGrossMarginResponse = negOperationGrossmarginApiResponse( kpiResponse);
        let mappedNegOperationTimeResponse = negOperationTimeApiResponse(i18n, kpiResponse);
        let mappedNegAverageSpread = negOperationAvrgSpreadApiResponse(i18n , kpiResponse);
        setNegOperationGrossMargin(mappedNegGrossMarginResponse)
        setNegOperationTime(mappedNegOperationTimeResponse)
        setNegOperationAvrgSpread(mappedNegAverageSpread)
       }
    })
    .catch((error) => {
      setKpiResponse(false)
      seterrorThrown(error)
     })
    setUnitOperationData(unitOperationMockData)
    setUnitOperationApiData(unitOperationApiMockData)
  }, [i18n])

  if (kpiResponse === true && Object.entries(unitOperationData && unitOperationApiData).length > 0) { 
    return (
      <>
        <PageHeader
          title={i18n.t('UNIT_OPERATION.TITLE')}
          description={i18n.t('UNIT_OPERATION.DESCRIPTION')}
        />
        <NegativeOperationGrossMargin negativeGrossMargin = {negOperationGrossMarginData}/>
        <AmountOfOperatingTime negOperationTime = {negOperationTimeData} />
        <AverageSpread negOperationAvrgSpread = {negOperationAvrgSpreadData} />
        <SpecificationSpread spreadOperation={spreadOperationMockData} time = {time} noDataHandler={noDataHandler}/>
        <ReferenceSpecificationOfSpreadOperation {...unitOperationData.referenceSpecification} />
        <SpreadNumberOfCases spreadBarData={unitOperationData.spreadBarChart}/>
        <Top20DurationOfSpread {...unitOperationData.top20DurationOfSpread} />
      </>
    );
  } else if (errorThrown != null) {
    return <LoadingError />;
  } else {
    return <Spinner />;
  }
};
export default UnitOperation;
