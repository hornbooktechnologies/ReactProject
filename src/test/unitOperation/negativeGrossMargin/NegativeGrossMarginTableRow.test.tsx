import React from "react";
import renderer from "react-test-renderer";
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";
import { NegGrossmarginTableRowModel } from "../../../models/unitOperation/NegOperationGrossmarginModel";
import NegativeGrossMarginTableRow from "../../../components/unitOperation/NegativeOperationGrossMargin/NegativeGrossMarginTableRow";

describe("Component: NegativeGrossMarginTableRow", () => {
  it("NegativeGrossMarginTableRow renders correctly", () => {
    const mockData: NegGrossmarginTableRowModel = {
      index: 1,
      prefix: "Â¥",
      suffix: "Oku",
      negativeGrossMarginRowData: {
        actualOrForcast: 20,
        customClass: "uk-tooltip-bottom-custom",
        header: "Annual",
        impactOnEBITDA: 9,
        plan: 31,
        previousActualOrForcast: 20,
        previousImpactOnEBITDA: 10,
        previousPlan: 30,
      },
    };
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <NegativeGrossMarginTableRow {...mockData} />
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
