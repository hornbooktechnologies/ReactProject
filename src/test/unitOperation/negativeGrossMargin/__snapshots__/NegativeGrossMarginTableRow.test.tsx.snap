// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: NegativeGrossMarginTableRow NegativeGrossMarginTableRow renders correctly 1`] = `
Array [
  <tr>
    <th
      className="bg-grey-blue-40-a10"
      colSpan={2}
    >
      UNIT_OPERATION.IMPACT_SPREAD_EBITDA
    </th>
    <td
      className="  uk-text-middle"
    >
      <span
        className="color-grey-blue-60 font-12"
      >
        (
         - 
        )
      </span>
      <span
        className="  font-14 uk-display-block"
      >
         
      </span>
    </td>
    <td
      className="  uk-text-middle"
    >
      <span
        className="color-grey-blue-60 font-12"
      >
        (
         - 
        )
      </span>
      <span
        className="  font-14 uk-display-block"
      >
         
      </span>
    </td>
    <td
      className="  uk-text-middle"
    >
      <span
        className="color-grey-blue-60 font-12"
      >
        (
         - 
        )
      </span>
      <span
        className="  font-14 uk-display-block"
      >
         
      </span>
    </td>
  </tr>,
  <tr>
    <th
      className="bg-grey-blue-40-a10 w-220"
      rowSpan={2}
    >
      UNIT_OPERATION.GROSSMARGIN_SPREAD
    </th>
    <th
      className="bg-grey-blue-40-a10 w-220"
    >
      LABELS.PLAN
    </th>
    <td
      className=" uk-text-middle"
    >
      <span
        className=" uk-display-block"
      >
         
      </span>
    </td>
    <td
      className=" uk-text-middle"
    >
      <span
        className=" uk-display-block"
      >
         
      </span>
    </td>
    <td
      className=" uk-text-middle"
    >
      <span
        className=" uk-display-block"
      >
         
      </span>
    </td>
  </tr>,
  <tr>
    <th
      className="bg-grey-blue-40-a10"
    >
      LABELS.ACTUAL_FORECAST
    </th>
    <td
      className=" uk-text-middle"
    >
      <span
        className=" uk-display-block"
      >
         
      </span>
    </td>
    <td
      className=" uk-text-middle"
    >
      <span
        className=" uk-display-block"
      >
         
      </span>
    </td>
    <td
      className=" uk-text-middle"
    >
      <span
        className=" uk-display-block"
      >
         
      </span>
    </td>
  </tr>,
]
`;
